vyper/semantics/analysis/module.py

```diff
@@ -161,6 +161,10 @@ def _compute_reachable_set(fn_t: ContractFunctionT, path: list[ContractFunctionT
             message = " -> ".join([f.name for f in path])
             raise CallViolation(f"Contract contains cyclic function call: {message}")
 
+        if g == fn_t:
```

Comment:
can we not merge this with the previous branch on line 160?

Reply:
opted for a different cycle detection approach: https://github.com/vyperlang/vyper/pull/4200/commits/5ac3ad037b4dcc023b2941609a0506b908b49b3b